pkg/errors/errors.go

```diff
@@ -0,0 +1,158 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"runtime"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Kind enums
+const (
+	KindNotFound   = http.StatusNotFound
+	KindBadRequest = http.StatusBadRequest
+	KindUnexpected = http.StatusInternalServerError
+)
+
+// Error is an Athens system error.
+// It carries information and behavior
+// as to what caused this error so that
+// callers can implement logic around it.
+type Error struct {
+	// Kind categories Athens errors into a smaller
+	// subset of errors. This way we can generalize
+	// what an error really is: such as "not found",
+	// "bad request", etc. The official categories
+	// are HTTP status code but the ones we use are
+	// imported into this package.
+	Kind     int
+	Op       Op
+	Module   M
+	Version  V
+	Err      error
+	Severity logrus.Level
+}
+
+// Error returns the underlying error's
+// string message. The logger takes care
+// of filling out the stack levels and
+// extra information.
+func (e Error) Error() string {
+	return e.Err.Error()
+}
+
+// Op describes any independent function or
+// method in Athens. A series of operations
+// forms a more readable stack trace.
+type Op string
+
+// M represents a module in an error
+// this is so that we can distinguish
+// a module from a regular error string or version.
+type M string
+
+// V represents a module version in an error
+type V string
+
+// E is a helper function to construct an Error type
+// Operation always comes first, module path and version
+// come second, they are optional. Args must have at least
+// an error or a string to describe what exactly went wrong.
+// You can optionally pass a Logrus severity to indicate
+// the log level of an error based on the context it was constructed in.
+func E(op Op, args ...interface{}) error {
+	e := Error{Op: op}
+	if len(args) == 0 {
+		msg := "errors.E called with 0 args"
+		_, file, line, ok := runtime.Caller(1)
+		if ok {
+			msg = fmt.Sprintf("%v - %v:%v", msg, file, line)
+		}
+		e.Err = errors.New(msg)
+	}
+	for _, a := range args {
+		switch a := a.(type) {
+		case error:
+			e.Err = a
+		case string:
+			e.Err = errors.New(a)
+		case M:
+			e.Module = a
+		case V:
+			e.Version = a
+		case logrus.Level:
+			e.Severity = a
+		case int:
```

Comment:
should we have a `default` here?

Reply:
@arschles I don't think so, but I'm happy either way. I'm not sure what the default would do here, the only thing it can do is either Panic or log "unknown type" which means we'd have to pass a logger to every error or just use fmt.Println which i think we should avoid. wdyt?

Reply:
I'm with you! Maybe we'll figure out what to put in a default here after this thing gets some more usage, but for now I'm good leaving it out